sysl2/sysl/integration/deps.go

```diff
@@ -0,0 +1,304 @@
+						!hasPattern("human", utils.MakeStrSetFromSpecificAttr("patterns", app.GetAttrs()).ToSlice()) {
+						r = append(r, item)
+					}
+					return true
+				}
+				if !hasPattern("human", utils.MakeStrSetFromSpecificAttr("patterns", app.GetAttrs()).ToSlice()) {
+					r = append(r, item)
+				}
+				return true
+			})
+		}
+		return true
+	})
+
+	return utils.MakeStrSet(r...)
+}
+
+func walkPassthrough(excludes, passthroughs utils.StrSet, dep *AppDependency, integrations *DependencySet, module *sysl.Module) {
+	target := dep.Target
+	targetName := target.Name
+	targetEndpoint := target.Endpoint
+	excludedApps := utils.MakeStrSet(targetName).Intersection(excludes)
+	undeterminedEps := utils.MakeStrSet(targetEndpoint).Intersection(utils.MakeStrSet(".. * <- *", "*"))
+	//Add to integration array since all dependencies are determined.
+	if len(excludedApps) == 0 && len(undeterminedEps) == 0 {
+		integrations.Deps[dep] = struct{}{}
+	}
+
+	// find the next outbound dep
+	if passthroughs.Contains(targetName) {
+		cs := NewCallSlice()
+		cs.CollectCallStatements(module.GetApps()[targetName].GetEndpoints()[targetEndpoint].GetStmt())
+		for _, call := range cs.GetSlice() {
+			nextAppName := strings.Join(call.GetTarget().GetPart(), " :: ")
+			nextEpName := call.GetEndpoint()
+			next := MakeAppElement(nextAppName, nextEpName)
+			nextDep := MakeAppDependency(target, next)
+			walkPassthrough(excludes, passthroughs, nextDep, integrations, module)
+		}
+	}
+}
+
+func (ds *DependencySet) buildArray() []*AppDependency {
+	r := []*AppDependency{}
+	for k := range ds.Deps {
+		r = append(r, k)
+	}
+
+	return r
+}
+
+func (ds *DependencySet) FindIntegrations(apps, excludes, passthroughs utils.StrSet, module *sysl.Module) *DependencySet {
+	integrations := NewDependencySet()
+	outboundDeps := NewDependencySet()
+	lenPassthroughs := len(passthroughs)
+	for dep := range ds.Deps {
+		appNames := dep.extractAppNames()
+		endpoints := dep.extractEndpoints()
+		isSubsection := isSub(appNames, apps)
+		isSelfSubsection := isSub(utils.MakeStrSet(dep.Self.Name), apps)
+		isTargetSubsection := isSub(utils.MakeStrSet(dep.Target.Name), passthroughs)
+		interExcludes := appNames.Intersection(excludes)
+		interEndpoints := endpoints.Intersection(utils.MakeStrSet(".. * <- *", "*"))
+		lenInterExcludes := len(interExcludes)
+		lenInterEndpoints := len(interEndpoints)
+		if isSubsection && lenInterExcludes == 0 && lenInterEndpoints == 0 {
+			integrations.Deps[dep] = struct{}{}
+		}
+		// collect outbound dependencies
+		if lenPassthroughs > 0 &&
+			((isSubsection || (isSelfSubsection && isTargetSubsection)) && lenInterExcludes == 0 && lenInterEndpoints == 0) {
+			outboundDeps.Deps[dep] = struct{}{}
+		}
+	}
+	if lenPassthroughs > 0 {
+		for dep := range outboundDeps.Deps {
+			walkPassthrough(excludes, passthroughs, dep, integrations, module)
+		}
+	}
+
+	return integrations
+}
+
+func (ds *DependencySet) ResolveDependencies(module *sysl.Module) {
+	apps := module.GetApps()
```

Comment:
all these checks were already done in 
```go
parse.go:func postProcess(mod *sysl.Module) {
```
now you are doing it again here.

Reply:
So I asked before if I could just merge the code here to build the dependencies with return a slice of dependency in `postProcess `.

Reply:
yep, we removed the call validation and only handle the dependencies for the specific application.

Reply:
Here removed the check logic and remain the collect dependency logic.